jans-cedarling/cedarling/src/authz/entities/create.rs

```diff
@@ -314,56 +318,59 @@ fn get_expression(
 
         let uid = EntityUid::from_type_name_and_id(
             EntityTypeName::from_str(entity_full_type_name.as_str()).map_err(|err| {
-                CedarPolicyCreateTypeError::EntityTypeName(entity_full_type_name.to_string(), err)
+                CreateCedarEntityError::EntityTypeName(entity_full_type_name.to_string(), err)
             })?,
-            EntityId::new(token_claim_value.as_str()?),
+            EntityId::new(token_claim.as_str()?),
         );
         RestrictedExpression::new_entity_uid(uid)
     };
     Ok(restricted_expression)
 }
 
 /// Build [`RestrictedExpression`] based on token_claim_value.
-/// It tries to find mapping and apply it to `token_claim_value` json value.
+/// It tries to find mapping and apply it to `token_claim` json value.
 fn get_record_expression(
     record: &CedarSchemaRecord,
     cedar_record_type: &EntityParsedTypeName<'_>,
-    token_claim_value: &Payload<'_>,
+    token_claim: &TokenClaim,
     schema: &CedarSchemaJson,
     claim_mapping: &ClaimMappings,
-) -> Result<RestrictedExpression, CedarPolicyCreateTypeError> {
-    // map json value of `token_claim_value` to TokenPayload object (HashMap)
-    let mapped_claim: TokenPayload = match claim_mapping.get_mapping(
-        token_claim_value.get_key(),
-        &cedar_record_type.full_type_name(),
-    ) {
-        Some(m) => m.apply_mapping(token_claim_value.get_value()).into(),
-        // if we do not have mapping, and value is json object, return TokenPayload based on it.
-        // if value is not json object, return empty value
-        None => {
-            if let Some(map) = token_claim_value.get_value().as_object() {
-                TokenPayload::from_json_map(map.to_owned())
-            } else {
-                TokenPayload::default()
-            }
-        },
-    };
+) -> Result<RestrictedExpression, CreateCedarEntityError> {
+    // map json value of `token_claim` to TokenPayload object (HashMap)
+    let mapped_claim: TokenData =
+        match claim_mapping.get_mapping(token_claim.key(), &cedar_record_type.full_type_name()) {
+            Some(m) => m.apply_mapping(token_claim.value()).into(),
+            // if we do not have mapping, and value is json object, return TokenPayload based on it.
+            // if value is not json object, return empty value
+            None => {
+                if let Some(map) = token_claim.value().as_object() {
+                    TokenData::from_json_map(map.to_owned())
+                } else {
+                    TokenData::default()
+                }
+            },
+        };
 
     let mut record_restricted_exps = Vec::new();
 
     for (attribute_key, entity_attribute) in record.attributes.iter() {
         let attribute_type = entity_attribute.get_type()?;
 
-        let mapped_claim_value = mapped_claim.get_payload(attribute_key)?;
+        let mapped_claim_value =
+            mapped_claim
+                .get_claim(attribute_key)
```

Comment:
Much easier to return error instead of Option. Like it was in previous for `.get_payload`

Reply:
`get_claim` is in the `jwt` module. and that is inthe `authz` module. returning an error for `get_claim` will either require `jwt` module to import the error from the `authz` or creating another error type. Calling `.ok_or` is a more straightforward solution.